Sources/NIOHTTP1Client/main.swift

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import NIO
+import NIOHTTP1
+
+print("Please enter line to send to the server")
+let line = readLine(strippingNewline: true)!
+
+private final class HTTPEchoHandler: ChannelInboundHandler {
+    public typealias InboundIn = HTTPClientResponsePart
+    public typealias OutboundOut = HTTPClientRequestPart
+
+    public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+
+        let clientResponse = self.unwrapInboundIn(data)
+        
+        switch clientResponse {
+        case .head(let responseHead):
+            print("Received status: \(responseHead.status)")
+        case .body(var byteBuffer):
+            if let string = byteBuffer.readString(length: byteBuffer.readableBytes) {
+                print("Received: '\(string)' back from the server.")
+            } else {
+                print("Received the line back from the server.")
+            }
+        case .end:
+            print("Closing channel.")
+            context.close(promise: nil)
+        }
+    }
+
+    public func errorCaught(context: ChannelHandlerContext, error: Error) {
+        print("error: ", error)
+
+        // As we are not really interested getting notified on success or failure we just pass nil as promise to
+        // reduce allocations.
+        context.close(promise: nil)
+    }
+
+    public func channelActive(context: ChannelHandlerContext) {
```

Comment:
Should we guard against being added to an already-active channel here?

Reply:
So in *general* it is possible that a channel handler is added to a stream *after* that stream has fired `channelActive`. In this specific case it’s not possible in the current implementation, but certain changes (e.g. for higher level protocol negotiation) could make it possible.

The way this is detected is by checking whether the channel is active in `handlerAdded`. Of course, caution needs to be required to avoid double-activating. This may all be more complexity than we need in this example, but it’s a fairly easy mistake to make.

Reply:
Thanks. I am actually doing this one as a prequel to the WebSockets client, so that very helpful for that in particular. I am not sure we need it for this example though. It doesn't really have any error checking. My impression (perhaps incorrectly) is that these are just 'as simple as possible' examples to get people started with a pipeline that can expand upon. I am happy to add it if you think it is required, but I would imagine that, in that case, we should probably add it to all client examples.

Reply:
@weissi thoughts?

Reply:
@Lukasa / @tigerpixel I'd say it's fine here, we can always improve